services/activitylog/pkg/service/service.go

```diff
@@ -80,7 +163,11 @@ func New(opts ...Option) (*ActivitylogService, error) {
 		valService:       o.ValueClient,
 		lock:             sync.RWMutex{},
 		registeredEvents: make(map[string]events.Unmarshaller),
+		tp:               o.TraceProvider,
+		tracer:           o.TraceProvider.Tracer("github.com/opencloud-eu/opencloud/services/activitylog/pkg/service"),
+		parentIdCache:    cache,
```

Comment:
Is this cache really worth it? Aren't all of the Stat() calls answered from the NATs cache in the storageproviders anyway?

Reply:
Cached stats aren't super expensive, yes. But on the other hand it's one stat per uploaded file and parent (except for the direct parent) up to the space root. So uploading 500 files in "a/b/c/" would trigger 1000 superfluous gateway->storageprovider->nats roundtrips. And the in-memory ttl cache is really lightweight and straightforward, so... I would vote for keeping it.

Reply:
Fine with me. I just a bit cautious when it comes to more caches. They tend to cause funny side-effects. But I in this special case the risk it pretty low.